File: JS/otherAlgorithms/buildBridge.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim().split("\n")
    : `3
2 2
1 5
13 29`.split("\n");

const t = Number(input[0]);

for (let i = 1; i < t + 1; i++) {
  const [n, m] = input[i].split(" ").map((num) => Number(num));
  let top = 1;
  let bottom = 1;

  const max = Math.max(m - n, n);
  const min = Math.min(m - n, n);

  for (let j = m; j > max; j--) {
    top *= j;
  }

  for (let j = 1; j < min + 1; j++) {
    bottom *= j;
  }

  console.log(parseInt(top / bottom));
}

File: src/otherAlgorithms/BuildBridge.java
package otherAlgorithms;

import java.io.*;

public class BuildBridge {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());

        for(int i = 0; i < t; i++){
            String[] nm = br.readLine().split(" ");
            int n = Integer.parseInt(nm[0]), m = Integer.parseInt(nm[1]);

            long top = 1, bottom = 1;

            int max = Math.max(m - n, n), min = Math.min(m - n, n);

            for(int j = m; j > max; j--){
                top *= j;
            }

            for(int j = 1; j < min + 1; j++){
                bottom *= j;
            }

            bw.write(String.valueOf(top / bottom) + "\n");
        }

        bw.flush();
        bw.close();
    }
}
